### PR TITLE
GH-43434: [Java][CI] Improve java-jars runtime

### PR DESCRIPTION
**Pull request by danepitkin**
### Rationale for this change

Improve Ubuntu C++ build / java-jars runtime, which currently takes 3+ hours.

### What changes are included in this PR?

* Use nuget to cache vcpkgs

### Are these changes tested?

CI

### Are there any user-facing changes?

No
* GitHub Issue: apache/arrow-java#457
* GitHub Issue: #457

### CODE REVIEW

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

File: ci/docker/java-jni-manylinux-201x.dockerfile [hunk 1]
```diff
@@ -18,23 +18,9 @@
...
-        --x-install-root=${VCPKG_ROOT}/installed \
-        --x-manifest-root=/arrow/ci/vcpkg \
-        --x-feature=dev \
-        --x-feature=flight \
-        --x-feature=gcs \
-        --x-feature=json \
-        --x-feature=parquet \
-        --x-feature=gandiva \
-        --x-feature=s3
-
```
Comment (assignUser) [inline ci/docker/java-jni-manylinux-201x.dockerfile:34]: This should remain in the image as people use this for local development too. 

But my understanding is that if you install the right triplet outside docker and mount it in the container for --x-install-root this should be a noop.

File: dev/tasks/python-wheels/github.osx.yml [hunk 2]
```diff
@@ -85,18 +85,21 @@ jobs:
...
+          --clean-after-build \
+          --x-install-root=${VCPKG_ROOT}/installed \
+          --x-manifest-root=/arrow/ci/vcpkg \
+          --x-feature=dev \
+          --x-feature=flight \
+          --x-feature=gcs \
+          --x-feature=json \
+          --x-feature=parquet \
+          --x-feature=gandiva \
+          --x-feature=s3
```
Comment (kou) [inline dev/tasks/python-wheels/github.osx.yml:102]: Could you revert needless indent changes?

File: dev/tasks/java-jars/github.yml [hunk 3]
```diff
@@ -50,6 +52,45 @@ jobs:
...
+            vcpkg install \
+              --clean-after-build \
+              --x-install-root=${VCPKG_ROOT}/installed \
+              --x-manifest-root=arrow/ci/vcpkg \
+              --x-feature=azure \
+              --x-feature=flight \
+              --x-feature=gcs \
+              --x-feature=json \
+              --x-feature=parquet \
+              --x-feature=s3
```
Comment (kou) [inline dev/tasks/java-jars/github.yml:93]:
```suggestion
              --x-feature=s3

```

**Comment by danepitkin** [comment]
> ```
> Failed to render template `java-jars/github.yml` with UndefinedError: 'secrets' is undefined
> The Archery job run can be found at: https://github.com/apache/arrow/actions/runs/10117085891
> ```

@assignUser Does infra need to set this job up with `secrets` in order to run?

**Comment by assignUser** [comment]
@danepitkin no that's an archery render fail. you need to escape gh macros like `${{ ... }}`  in jinja templates like this `{{ '${{...}}' }}` you can test it with `archery crossbow render java-jars`

**Comment by danepitkin** [comment]
> @danepitkin no that's an archery render fail. you need to escape gh macros like `${{ ... }}` in jinja templates like this `{{ '${{...}}' }}` you can test it with `archery crossbow render java-jars`

Ahh I see. Thank you!

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
Ubuntu aarch64 is having trouble installing vcpkg. It can't find a prebuilt version and then fails trying to build from source:
```
Unable to determine a binary release of vcpkg; attempting to build from source.
Downloading vcpkg tool sources
Building vcpkg-tool...
./bootstrap-vcpkg.sh: 1: eval: cmake: not found
```
https://github.com/ursacomputing/crossbow/actions/runs/10204097563/job/28231825382

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
@github-actions crossbow submit java-jars

**Comment by danepitkin** [comment]
I am out of my depth here and don't have the time to dedicate to learning it right now.

**Pull request closed without merging**